packages/algjulia-service/src/decapodes.jl

```diff
@@ -226,86 +297,44 @@ struct SimResult
 end
 export SimResult
 
-function SimResult(sol::ODESolution, mesh::EmbeddedDeltaDualComplex2D)
+function SimResult(sol::ODESolution, system::System)
+
+    points = collect(values(system.mesh.subparts.point.m));
 
-    points = collect(values(mesh.subparts.point.m));
+    xlen = 51; ylen = 51;
+
+    statevars = infer_state_names(system.pode)
+    statevar = length(statevars) == 1 ? first(statevars) : error("$statevars must be length one")
```

Comment:
Seems like the `System` struct should store the list of statevars (or the unique state variable, if that is being enforced) so it doesn't have to be inferred twice.

Reply:
Makes sense!